packages/wallet/wdk/src/sequence/wallets.ts

```diff
@@ -446,6 +447,26 @@ function toModulesTopology(modules: Config.SapientSignerLeaf[]): Config.Topology
     } as Config.SignerLeaf
   }
 
+  // If we have a guard, we need to add it as a signer to each of the modules
+  // (excluding recovery)
+  if (guardTopology) {
+    const leaves = modules.map((module) => {
+      // Leave out the recovery module (which has weight 255)
+      if (module.weight === 255n) {
+        return module
+      }
+
+      return {
+        type: 'nested',
+        weight: 255n,
```

Comment:
I don't think this is right. This gives any (module + guard) pair full access to the wallet.

Reply:
With only the sessions module, this works as expected, as (sessions + guard) should be able to access the wallet. But it is gonna break for modules that aren't supposed to have such power. I'll get it changed so that we only nest the sessions and leave other modules as-is

Reply:
Changed the type of modules from `SapientSignerLeaf` to a new `Module` type that includes guard information. This way we can specify which modules need it and which don't.